apps/src/code-studio/components/progress/MiniView.jsx

```diff
@@ -0,0 +1,31 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import CourseProgress from './CourseProgress';
+
+/**
+ * The course progress dropdown you get when you click the arrow in the header.
+ */
+const MiniView = React.createClass({
+  propTypes: {
+    hasFullProgress: PropTypes.bool.isRequired
+  },
+
+  render() {
+    const { hasFullProgress } = this.props;
+
+    if (!hasFullProgress) {
+      // Ideally we would specify inline CSS instead of using a classname here,
+      // but the image used here gets digested by rails, and we don't know the
+      // digested path
+      return <div className="loading"/>;
```

Comment:
Really don't particularly like this. Might be worth either (a) using a FontAwesome spinner or (b) moving this asset into apps so that we don't have to depend on application.scss (https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/assets/stylesheets/application.scss#L824).